# Basic SQL Queries: TMDb Database
© Explore Data Science Academy

## Instructions to Students

This challenge is designed to determine how much you have learned so far and will test your knowledge on Basic SQL queries.

The answers for this challenge should be selected on Athena for each corresponding Multiple Choice Question. The questions are included in this notebook and are numbered according to the Athena Questions. The options to choose from for each question has also been included.

*NOTE:*
Do not add or remove cells in this notebook. Do not edit or remove the `%%sql` comment as it is required to run each cell.

**_Good Luck!_**

## Honour Code

I YOUR NAME, YOUR SURNAME, confirm - by submitting this document - that the solutions in this notebook are a result of my own work and that I abide by the EDSA honour code (https://drive.google.com/file/d/1QDCjGZJ8-FmJE3bZdIQNwnJyQKPhHZBn/view?usp=sharing).

Non-compliance with the honour code constitutes a material breach of contract.

## The TMDb Database

In this challenge you will be exploring the [The Movie Database](https://www.themoviedb.org/): an online movie and TV show database, which houses some of the most popular movies and TV shows at your finger tips. The TMDb database supports 39 official languages used in over 180 countries daily and dates all the way back to 2008. 


<img src="https://github.com/Explore-AI/Pictures/blob/master/sql_tmdb.jpg?raw=true" width=80%/>


Below is an Entity Relationship Diagram (ERD) of the TMDb database:

<img src="https://github.com/Explore-AI/Pictures/blob/master/TMDB_ER_diagram.png?raw=true" width=70%/>

As can be seen from the ER diagram, the TMDb database consists of `12 tables` containing information about movies, cast, genre and so much more.  

For this challenge we will only be making use of the Movies table from the TMDb database. We will be applying the use of basic SQL queries to gain insightful information from the Movies table. The Movie table consists of: 
- 15 columns
- 4804 rows
- Includes information on movie revenue, release date, popularity and so much more ! 

Let's get started!

## Loading the database

To begin and start making use of SQL queries you need to prepare your SQL environment you can do this by loading in the magic command `%load_ext sql`, next you can go ahead and load in your database. To do this you will need to ensure you have downloaded the `TMDB.db`sqlite file from Athena and have stored it in a known location. Now that you have all the prerequistes you can go ahead and load it into the notebook. 

In [1]:
%load_ext sql

In [2]:
%%sql 

sqlite:///TMDB.db

## Questions on Basic SQL Queries 

Use the given cell below each question to execute your SQL queries to find the correct input from the options provided for the multiple choice questions on Athena.

To attempt this you will use what you have learned about the following functions and statements:

- SELECT
- FROM
- WHERE
- LIKE
- COUNT
- OR
- AND
- BETWEEN

**Question 1**

What is the SQL code to see the whole movies table?

**Options:** 
- SELECT ALL FROM Movies
- SELECT TABLE FROM Movies
- SELECT * FROM Movies
- SELECT % FROM Movies

In [10]:
%%sql 

SELECT * FROM Movies
LIMIT 10;

 * sqlite:///TMDB.db
Done.


[(5, 'Four Rooms', '1995-12-09 00:00:00.000000', 4000000, None, 'en', 'Four Rooms', "It's Ted the Bellhop's first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments. It seems that this evening's room service is serving up one unbelievable happening after another.", 22.87623, 4300000.0, 98.0, 'Released', "Twelve outrageous guests. Four scandalous requests. And one lone bellhop, in his first day on the job, who's in for the wildest New year's Eve of his life.", 6.5, 530),
 (11, 'Star Wars', '1977-05-25 00:00:00.000000', 11000000, 'http://www.starwars.com/films/star-wars-episode-iv-a-new-hope', 'en', 'Star Wars', 'Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker a ... (14 characters truncated) ... tain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.', 126.393695, 775398007.0, 121.0, 'Released', 'A long time ago in a galaxy far, far away...', 8.1, 6624),
 (12, 'Finding Nemo', '2003-05-30 00:00:00.000000', 94000000, 'http://movies.disney.com/finding-nemo', 'en', 'Finding Nemo', "Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome ... (35 characters truncated) ... orgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.", 85.688789, 940335536.0, 100.0, 'Released', "There are 3.7 trillion fish in the ocean, they're looking for one.", 7.6, 6122),
 (13, 'Forrest Gump', '1994-07-06 00:00:00.000000', 55000000, None, 'en', 'Forrest Gump', "A man with a low IQ has accomplished great things in his life and been present during significant historic events - in each case, far exceeding what  ... (108 characters truncated) ... rest Gump' is the story of a man who rose above his challenges, and who proved that determination, courage, and love are more important than ability.", 138.133331, 677945399.0, 142.0, 'Released', "The world will never be the same, once you've seen it through the eyes of Forrest Gump.", 8.2, 7927),
 (14, 'American Beauty', '1999-09-15 00:00:00.000000', 15000000, 'http://www.dreamworks.com/ab/', 'en', 'American Beauty', "Lester Burnham, a depressed suburban father in a mid-life crisis, decides to turn his hectic life around after developing an infatuation with his daughter's attractive friend.", 80.878605, 356296601.0, 122.0, 'Released', 'Look closer.', 7.9, 3313),
 (16, 'Dancer in the Dark', '2000-05-17 00:00:00.000000', 12800000, None, 'en', 'Dancer in the Dark', "Selma, a Czech immigrant on the verge of blindness, struggles to make ends meet for herself and her son, who has inherited the same genetic disorder  ... (121 characters truncated) ... her love of musicals, escaping life's troubles - even if just for a moment - by dreaming up little numbers to the rhythmic beats of her surroundings.", 22.022228, 40031879.0, 140.0, 'Released', "You don't need eyes to see.", 7.6, 377),
 (18, 'The Fifth Element', '1997-05-07 00:00:00.000000', 90000000, None, 'en', 'The Fifth Element', 'In 2257, a taxi driver is unintentionally given the task of saving a young girl who is part of the key that will ensure the survival of humanity.', 109.528572, 263920180.0, 126.0, 'Released', 'There is no future without it.', 7.3, 3885),
 (19, 'Metropolis', '1927-01-10 00:00:00.000000', 92620000, None, 'de', 'Metropolis', "In a futuristic city sharply divided between the working class and the city planners, the son of the city's mastermind falls in love with a working class prophet who predicts the coming of a savior to mediate their differences.", 32.351527, 650422.0, 153.0, 'Released', 'There can be no understanding between the hands and the brain unless the heart acts as mediator.', 8.0, 657),
 (20, 'My Life W

**Question 2**

What was the budget for the movie “Inception”?

**Options:**
- $224 000 000

- $160 000 000

- $344 000 000

- $115 000 000


In [6]:
%%sql 
SELECT budget FROM Movies
WHERE title = "Inception";

 * sqlite:///TMDB.db
Done.


[(160000000,)]

**Question 3**

What is the runtime of the movie "Titanic"?

**Options:**
- 146
- 158
- 122
- 194

In [6]:
%%sql 
SELECT runtime FROM Movies
WHERE title = "Titanic";

 * sqlite:///TMDB.db
Done.


[(194.0,)]

**Question 4**

How many movies do not have English as their original language? (Hint: “en” is the abbreviation for English)

**Options:**
- 492
- 298
- 387
- 315

In [13]:
%%sql 
SELECT COUNT(*) FROM Movies
WHERE original_language != "en";

 * sqlite:///TMDB.db
Done.


[(298,)]

**Question 5**

How many movies are there that have a popularity score of more than 250?

**Options:**
- 7
- 5
- 9
- 11

In [14]:
%%sql 

SELECT COUNT(*) FROM Movies
WHERE popularity >= 250;

 * sqlite:///TMDB.db
Done.


[(7,)]

**Question 6**

 How many movies are there where the title is not the same as the original title?
 
 **Options:**
 -  187
 - 261
 - 74
 - 24

In [15]:
%%sql
SELECT COUNT(*) FROM Movies
WHERE original_title != title;

 * sqlite:///TMDB.db
Done.


[(261,)]

**Question 7**

 How many movies are there that managed to get a popularity rating of more than 100 with a budget of less than $10 000 000?
 
**Options:**
 - 11
 - 18
 - 5
 - 15

In [17]:
%%sql 
SELECT COUNT(*) FROM Movies
WHERE popularity >= 100 AND budget <= 10000000;

 * sqlite:///TMDB.db
Done.


[(5,)]

**Question 8** 

How many movies are there that have the word ‘love’ anywhere in the title? (Hint: The L in the word love can be upper or lower case and can be included in words such as ‘lovers’.)

**Options:**
- 67
- 58
- 71
- 49

In [16]:
%%sql

SELECT COUNT(*) FROM Movies
WHERE title LIKE '%love%';

 * sqlite:///TMDB.db
Done.


[(71,)]

**Question 9**

How many movies were released between the dates 1 August 2012 and 31 July 2013?  

**Options:**
- 227 
- 295
- 3
- 208

In [7]:
%%sql 

SELECT COUNT(*) FROM Movies
WHERE release_date BETWEEN "2012-08-01 00:00:00.000000" AND "2013-07-31 00:00:00.000000";

 * sqlite:///TMDB.db
Done.


[(227,)]

**Question 10**

You have had a long day and want to sit back and enjoy a movie.  Unfortunately, today you are only in the mood for a very specific type of movie. It definitely needs to be in English. It should also be new, something after 1 Jan 2010, but not too new as you might have seen it recently, so it must have been released before 1 Jan 2016. It should also be a romantic movie, so make sure it has the word love somewhere in the title.  You also want it to be a big blockbuster, so the budget of the move must be more than $10 000 000.

What is the movie with the highest popularity that meets all of your requirements?

**Options:**
-  Love & Other Drugs
- From Paris with Love
-  Crazy, Stupid, Love
- Eat Pray Love

In [46]:
%%sql

SELECT * FROM Movies WHERE release_date
BETWEEN "2010-01-01 00:00:00.000000" AND "2016-01-01 00:00:00.000000" AND
title LIKE "%love%" And budget >= 10000000 AND original_language = "en";


 * sqlite:///TMDB.db
Done.


[(26389, 'From Paris with Love', '2010-02-05 00:00:00.000000', 52000000, 'http://www.frompariswithlovefilm.com/', 'en', 'From Paris with Love', "James Reese has a good job as an ambassador's aid in France, but his real passion is a side gig, working in a minor role in the CIA. He would love to ... (149 characters truncated) ...  desk job, but when he learns that the same guys they're trying to catch are after him, James realizes that Charlie may be his only hope of survival.", 27.916284, 52826594.0, 92.0, 'Released', 'Two agents. One city. No merci.', 6.1, 675),
 (38167, 'Eat Pray Love', '2010-08-12 00:00:00.000000', 60000000, 'http://www.sonypictures.com/movies/eatpraylove/', 'en', 'Eat Pray Love', 'Liz Gilbert had everything a modern woman is supposed to dream of having û a husband, a house and a successful career û yet like so many others, she  ... (317 characters truncated) ... sure of nourishment by eating in Italy, the power of prayer in India and, finally and unexpectedly, the inner peace and balance of true love in Bali.', 28.800112, 204594016.0, 133.0, 'Released', 'Let Yourself GO', 5.8, 636),
 (43347, 'Love & Other Drugs', '2010-11-22 00:00:00.000000', 30000000, 'http://www.loveandotherdrugsthemovie.com/', 'en', 'Love & Other Drugs', "Maggie, an alluring free spirit who won't let anyone - or anything - tie her down. But she meets her match in Jamie, whose relentless and nearly infa ... (88 characters truncated) ...  sales. Maggie and Jamie's evolving relationship takes them both by surprise, as they find themselves under the influence of the ultimate drug: love.", 20.236951, 102820008.0, 112.0, 'Released', 'Addicted to one-night stands or dependent on one another?', 6.6, 1233),
 (50646, 'Crazy, Stupid, Love.', '2011-07-29 00:00:00.000000', 50000000, 'http://crazystupidlove.warnerbros.com/index.html', 'en', 'Crazy, Stupid, Love.', "Cal Weaver is living the American dream. He has a good job, a beautiful house, great children and a beautiful wife, named Emily. Cal's seemingly perf ... (139 characters truncated) ... n the fickle world of dating. Enter, Jacob Palmer, a self-styled player who takes Cal under his wing and teaches him how to be a hit with the ladies.", 37.990705, 142851197.0, 118.0, 'Released', 'This Is Stupid', 7.0, 2443),
 (79698, 'The Lovers', '2015-02-13 00:00:00.000000', 27000000, None, 'en', 'The Lovers', 'The Lovers is an epic romance time travel adventure film. Helmed by Roland JoffT from a story by Ajey Jhankar, the film is a sweeping tale of an impo ... (146 characters truncated) ... British officer in 18th century colonial India, the Indian woman he falls deeply in love with, an American present-day marine biologist and his wife.', 2.418535, 0.0, 109.0, 'Released', 'Love is longer than life.', 4.8, 34),
 (81836, 'To Rome with Love', '2012-04-20 00:00:00.000000', 17000000, None, 'en', 'To Rome with Love', 'Four tales unfold in the Eternal City: While vacationing in Rome, architect John encounters a young man whose romantic woes remind him of a painful i ... (110 characters truncated) ... eizes the opportunity to rejuvenate his own flagging career; a young couple have separate romantic interludes; a spotlight shines on an ordinary man.', 15.013386, 73244881.0, 111.0, 'Released', None, 5.6, 546),
 (226857, 'Endless Love', '2014-02-12 00:00:00.000000', 20000000, None, 'en', 'Endless Love', "A privileged girl and a charismatic boy's instant desire sparks a love affair made only more reckless by parents trying to keep them apart.", 27.256849, 34077920.0, 103.0, 'Released', 'Say Goodbye to Innocence', 6.7, 542),
 (316002, 'Accidental Love', '2015-02-10 00:00:00.000000', 26000000, None, 'en', 'Accidental Love', 'A small town waitress gets a nail accidentally lodged in her head causing unpredictable behavior that leads her to Washington, DC.  Sparks fly when she meets a clueless young senator who takes up her cause - but what happens when love interferes with what you stand for?', 5.644844, 0.0, 100.0, 'Re